File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/ImageMetaData.java
package quintype.com.templatecollectionwithrx.models.story;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.Arrays;

/**
 * An object instance representing Story ImageMetaData
 *
 * @author Imran imran@quintype.com
 * @author Madhu madhu@quintype.com
 */
public class ImageMetaData implements Parcelable {

    @SerializedName("width")
    private int width;
    @SerializedName("height")
    private int height;
    @SerializedName("focus-point")
    private int[] focusPoints;

    @Override
    public String toString() {
        return "ImageMetaData{" +
                "width=" + width +
                ", height=" + height +
                ", focusPoints=" + Arrays.toString(focusPoints) +
                '}';
    }

    public int width() {
        return width;
    }

    public int height() {
        return height;
    }

    public int[] focusPoints() {
        return focusPoints;
    }

    public void focusPoints(int[] focusPoints) {
        this.focusPoints =focusPoints;
    }

    public ImageMetaData() {
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.width);
        dest.writeInt(this.height);
        dest.writeIntArray(this.focusPoints);
    }

    protected ImageMetaData(Parcel in) {
        this.width = in.readInt();
        this.height = in.readInt();
        this.focusPoints = in.createIntArray();
    }

    public static final Creator<ImageMetaData> CREATOR = new Creator<ImageMetaData>() {
        public ImageMetaData createFromParcel(Parcel source) {
            return new ImageMetaData(source);
        }

        public ImageMetaData[] newArray(int size) {
            return new ImageMetaData[size];
        }
    };
}

File: androidcore/src/main/java/com/example/androidcore/models/story/ImageMetaData.kt
package com.example.androidcore.models.story

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import java.util.*

/**
 * An object instance representing Story ImageMetaData
 * Created TemplateCollectionWithRx by rakshith on 9/4/18.
 */

class ImageMetaData : Parcelable {

    @SerializedName("width")
    var width: Int
    @SerializedName("height")
    var height: Int
    @SerializedName("focus-point")
    var focusPoints: IntArray? = null

    override fun toString(): String {
        return "ImageMetaData{" +
                "width=" + width +
                ", height=" + height +
                ", focusPoints=" + Arrays.toString(focusPoints) +
                '}'.toString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(this.width)
        dest.writeInt(this.height)
        dest.writeIntArray(this.focusPoints)
    }

    protected constructor(parcel: Parcel) {
        this.width = parcel.readInt()
        this.height = parcel.readInt()
        this.focusPoints = parcel.createIntArray()
    }

    companion object CREATOR : Parcelable.Creator<ImageMetaData> {
        override fun createFromParcel(source: Parcel): ImageMetaData {
            return ImageMetaData(source)
        }

        override fun newArray(size: Int): Array<ImageMetaData?> {
            return arrayOfNulls(size)
        }
    }
}
